templates/web/legacy/embedded_legacy_form.ftl

```diff
@@ -248,6 +254,12 @@
               action: 'cancelled'
             }, '*');
           },
+          pendingChanges: function() {
+            window.top.postMessage({
+              type: 'LEGACY_CODE_EDITOR_PENDING_CHANGES',
+              action: 'pendingChanges'
```

Comment:
As above.
```suggestion
              payload: 'pendingChanges'
```
These don't respect the type/payload pattern. Also, payload should be an object.